ts/packages/anchor/src/workspace.ts

```diff
@@ -1,108 +1,86 @@
-import camelCase from "camelcase";
 import * as toml from "toml";
-import { PublicKey } from "@solana/web3.js";
+import { snakeCase } from "snake-case";
 import { Program } from "./program/index.js";
-import { Idl } from "./idl.js";
 import { isBrowser } from "./utils/common.js";
 
-let _populatedWorkspace = false;
-
 /**
  * The `workspace` namespace provides a convenience API to automatically
  * search for and deserialize [[Program]] objects defined by compiled IDLs
  * in an Anchor workspace.
  *
  * This API is for Node only.
  */
-const workspace = new Proxy({} as any, {
-  get(workspaceCache: { [key: string]: Program }, programName: string) {
-    if (isBrowser) {
-      throw new Error("Workspaces aren't available in the browser");
-    }
+const workspace = new Proxy(
+  {},
+  {
+    get(workspaceCache: { [key: string]: Program }, programName: string) {
+      if (isBrowser) {
+        throw new Error("Workspaces aren't available in the browser");
+      }
+
+      // Converting `programName` to snake_case enables the ability to use any
+      // of the following to access the workspace program:
+      // `workspace.myProgram`, `workspace.MyProgram`, `workspace["my-program"]`...
+      programName = snakeCase(programName);
+
+      // Check whether the program name contains any digits
+      if (/\d/.test(programName)) {
+        // Numbers cannot be properly converted from camelCase to snake_case,
+        // e.g. if the `programName` is `myProgram2`, the actual program name could
+        // be `my_program2` or `my_program_2`. This implementation assumes the
+        // latter as the default and always converts to `_numbers`.
```

Comment:
We used `my_program2` and this is a pretty big breaking change since there's no way to override the name. We're now stuck having to rename our program everywhere

Reply:
I also prefer `my_program2` instead of `my_program_2` but we had examples using the other way:

https://github.com/coral-xyz/anchor/blob/d9a9f193947ec869e60eab58453f1791fff06576/examples/tutorial/basic-2/programs/basic-2/Cargo.toml#L10

This was a quick solution to fix CI errors but I think we should either support both ways or provide a good error message that explains how to fix the name related issues.